karma-base.js

```diff
@@ -3,17 +3,104 @@ var url = require('url');
 var fs = require('fs');
 var path = require('path');
 var notes_path = process.env.CTEST_NOTES_PATH || path.resolve('notes');
+var image_path = process.env.CTEST_IMAGE_PATH || path.resolve('images');
 var test_case = process.env.GEOJS_TEST_CASE || 'tests/all.js';
 var getRawBody = require('raw-body');
 
 // Create the notes directory, if it doesn't exist.
 if (!fs.existsSync(notes_path)) {
   fs.mkdirSync(notes_path);
 }
+if (!fs.existsSync(image_path)) {
+  fs.mkdirSync(image_path);
+}
 
 /**
- * Express style middleware to handle REST requests
- * to `/notes` on the test server.
+ * This function returns true if (1) there is an environment variable set
+ * called "TEST_SAVE_IMAGE", and either (2a) the name of the test appears in
```

Comment:
I should give credit to @ronichoudhury, as some of this I lifted from the [candela](https://github.com/Kitware/candela/) project.